### PR TITLE
Update stats

### PR DESCRIPTION
**Pull request by paulzierep**

### CODE REVIEW

File: bin/extract_galaxy_tools.py [hunk 1]
```diff
@@ -29,7 +30,6 @@
     "https://usegalaxy.org",
     "https://usegalaxy.org.au",
     "https://usegalaxy.eu",
-    "https://usegalaxy.fr",
```
Comment (bebatut) [inline bin/extract_galaxy_tools.py:32]: Why did you remove it?

Comment (bebatut) [reply bin/extract_galaxy_tools.py]: There is a list somewhere with public servers

**Comment by bebatut** [comment]
Can you explain a bit more in the PR what are the changes you are doing? Thanks

**Comment by paulzierep** [comment]
a) This PR updates the stats mapping (between EU and the tools) using the tool IDs `Galaxy tool ids` instead of the suite name `Galaxy wrapper id` -> tested on my branch, this now gets all stats for a tool provided they are installed on EU.

b) it also changes the tool availability in the results table. Each server gets one column where the number of tools installed on that server is listed, and there is also a column for `Tool count` number of tools in that suite. From the individual columns another column is generated which shows the availability of the servers i.e. `https://usegalaxy.org.au, https://usegalaxy.eu, https://usegalaxy.fr`

File: bin/extract_galaxy_tools.py [hunk 2]
```diff
@@ -520,9 +540,8 @@ def get_tool_count_per_server(tool_ids: Any) -> pd.Series:
         tool_id_list = [x.strip(" ") for x in tool_ids.split(",")]
         data = check_tools_on_servers(tool_id_list)
         result_df: pd.DataFrame = pd.DataFrame()
-        result_df["true_count"] = data.sum(axis=1).astype(str)
-        result_df["false_count"] = len(data.columns)
-        result_df["counts"] = result_df.apply(lambda x: "({}/{})".format(x["true_count"], x["false_count"]), axis=1)
+        result_df["counts"] = data.sum(axis=1).astype(str)
+        result_df.loc["Tool count", "counts"] = len(data.columns)
```
Comment (bebatut) [inline bin/extract_galaxy_tools.py:544]: `Tool count` might be a bit vague. And probably better after `Galaxy tool ids` list

File: bin/extract_galaxy_tools.py [hunk 3]
```diff
@@ -542,6 +561,32 @@ def add_instances_to_table(
...
+
+def aggregate_servers(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Aggregates a list of servers where the tools are installed
+
+    :param df: the results dataframe that already contains for each server a column
+    """
+
+    sub_df = df.loc[:, GALAXY_SERVER_URLS]
+    df["Server availability"] = sub_df.apply(get_server_list, axis=1)
```
Comment (bebatut) [inline bin/extract_galaxy_tools.py:586]: `Server availability` should probably before the individual server columns

File: bin/extract_galaxy_tools.py [hunk 4]
```diff
@@ -28,12 +28,19 @@
...
-    "https://usegalaxy.fr",
-]
+# GALAXY_SERVER_URLS = [
+#     "https://usegalaxy.org",
+#     "https://usegalaxy.org.au",
+#     "https://usegalaxy.eu",
+#     "https://usegalaxy.fr",
+# ]
+
+GALAXY_SERVER_URLS = {
```
Comment (bebatut) [inline bin/extract_galaxy_tools.py:38]: Maybe named it USEGALAXY_*_SERVER_URLS

**Comment by paulzierep** [comment]
Need to check if https://training.galaxyproject.org/api/#/tools/get_psl_json can improve that

File: bin/extract_galaxy_tools.py [hunk 5]
```diff
@@ -23,18 +23,45 @@
 from github.ContentFile import ContentFile
 from github.Repository import Repository
 
+COLUMN_ORDER = [
+    "Galaxy wrapper id",
```
Comment (bebatut) [inline bin/extract_galaxy_tools.py:27]: Should we not call it "Galaxy suite id"?

File: bin/extract_galaxy_tools.py [hunk 6]
```diff
@@ -23,18 +23,45 @@
...
+    "Conda version",
+    "Galaxy wrapper version",
+    "Status",
+    "ToolShed categories",
+    "ToolShed id",
+    "Source",
+    "Galaxy wrapper owner",
+    "Galaxy wrapper source",
+    "Galaxy wrapper parsed folder",
+    "Galaxy Star Availability",
```
Comment (bebatut) [inline bin/extract_galaxy_tools.py:48]: Why having a separate column here?

File: bin/extract_galaxy_tools.py [hunk 7]
```diff
@@ -23,18 +23,45 @@
...
+    "All Server Availability",
```
Comment (bebatut) [inline bin/extract_galaxy_tools.py:49]: Maybe "Public servers with at least one tool"

File: bin/extract_galaxy_tools.py [hunk 8]
```diff
@@ -23,18 +23,45 @@
...
+    "Tools available on: UseGalaxy.org",
+    "Tools available on: UseGalaxy.org.au",
+    "Tools available on: UseGalaxy.eu",
+    "Tools available on: UseGalaxy.org.fr",
```
Comment (bebatut) [inline bin/extract_galaxy_tools.py:53]:
```suggestion
    "Tools available on UseGalaxy.org",
    "Tools available on UseGalaxy.org.au",
    "Tools available on UseGalaxy.eu",
    "Tools available on UseGalaxy.fr",
```

File: bin/extract_galaxy_tools.py [hunk 9]
```diff
@@ -23,18 +23,45 @@
...
+    "No. of tool users (2022-2023) (usegalaxy.eu)",
```
Comment (bebatut) [inline bin/extract_galaxy_tools.py:54]:
```suggestion
    "Tool users in 2022-2023 on UseGalaxy.eu",
```

File: bin/extract_galaxy_tools.py [hunk 10]
```diff
@@ -23,18 +23,45 @@
...
+    "Total tool usage (usegalaxy.eu)",
```
Comment (bebatut) [inline bin/extract_galaxy_tools.py:55]:
```suggestion
    "Total tool usage on UseGalaxy.eu",
```

**Comment by paulzierep** [comment]
Done with: https://github.com/galaxyproject/galaxy_tool_metadata_extractor/pull/120 and https://github.com/galaxyproject/galaxy_tool_metadata_extractor/pull/113
